docs/diagrams/wallet-lifecycle/wallet-state-transition.tikz

```diff
@@ -0,0 +1,138 @@
+% Style of the labels is defined in `every node/.style`.
+
+\draw[->](unknown)      --  node[below right, pos=0, yshift=-5mm, align=left] {- registerNewWallet}                   (live);
+\draw[->](live)         --  node[below right, pos=0, yshift=-5mm, align=left] {- notifyWalletHeartbeatFailed\\
+                                                                               - notifyCloseableWallet\\
+                                                                               - notifyRedemptionTimeout}            (movingFunds);
+\draw[->](movingFunds)  --  node[below right, pos=0, yshift=-5mm, align=left] {- submitMovingFundsProof\\
+                                                                               - notifyMovingFundsBelowDust}         (closing);
+\draw[->](closing)      --  node[below right, pos=0, yshift=-5mm, align=left] {- notifyWalletClosingPeriodElapsed}   (closed);
+\draw[->](live)         -|  node[above right, pos=0, xshift=5mm, align=left]  {- notifyWalletHeartbeatFailed *\\
+                                                                               - notifyCloseableWallet *\\
+                                                                               - notifyRedemptionTimeout *\\
+                                                       \sffamily\footnotesize{(* only if there's no main UTXO)}}     +(10cm,0)
+                        |-                                                                                           (closing);
+\draw[->](live)         -|  node[above left, pos=0, xshift=-5mm, align=right] {- notifyFraudChallengeDefeatTimeout\\
+                                                                               - notifyMovedFundsSweepTimeout}       ([xshift=-50mm]terminated);
+\draw[->](movingFunds)  -|  node[above left, pos=0, xshift=-5mm, align=right] {- notifyFraudChallengeDefeatTimeout\\
+                                                                               - notifyMovingFundsTimeout\\
+                                                                               - notifyMovedFundsSweepTimeout}       (terminated);
+\draw[->](closing)      -|  node[above left, pos=0, xshift=-5mm, align=right] {- notifyFraudChallengeDefeatTimeout}  ([xshift=15mm]terminated);
+
+
+% TEXT BOX
+
+\node [box, below = of closed, rectangle, xshift=-10mm, yshift=10mm] (box_label)
+{FUNCTIONS AVAILABLE PER STATE\\ \hfill};
+
+\node [box, below of=box_label, rectangle split, rectangle split horizontal, rectangle split parts=3, yshift=-5mm,](box_content)
+{
+LIVE:\\
+  \ttfamily{
+  - revealDeposit\\
+  - submitDepositSweepProof\\
+  - submitMovedFundsSweepProof\\
+  - notifyMovedFundsSweepTimeout\\
+  - requestRedemption\\
+  - submitRedemptionProof\\
+  - submitFraudChallenge\\
+  - defeatFraudChallenge\\
+  - notifyFraudChallengeDefeatTimeout\\
+  - notifyWalletHeartbeatFailed\\
+  - notifyCloseableWallet\\
+  - notifyRedemptionTimeout}
```

Comment:
1. There is no `notifyWalletHeartbeatFailed` in the `Bridge`. Instead, the `WalletRegistry.notifyOperatorInactivity` where the heartbeat failure is received, calls `__ecdsaWalletHeartbeatFailedCallback` on the `Bridge`. Let's focus only on `Bridge` functions for this diagram to make it easier to follow and replace `notifyWalletHeartbeatFailed` with `__ecdsaWalletHeartbeatFailedCallback`. 

2. `defeatFraudChallenge` can be called in all states.

3. There is a function `defeatFraudChallengeWithHeartbeat` added recently to the contract that is not yet covered in diagrams. It can be called in all states.

4. There is a function `receiveBalanceApproval` added recently to the contract that that is not yet covered in diagrams.

5. Let's order functions in the same way as they are ordered in the `Bridge` contract.

Applying suggestions 1-5:
```suggestion
LIVE:\\
  \ttfamily{
  - revealDeposit\\
  - submitDepositSweepProof\\
  - requestRedemption\\
  - receiveBalanceApproval \\
  - submitRedemptionProof\\
  - notifyRedemptionTimeout\\
  - submitMovedFundsSweepProof\\
  - notifyMovedFundsSweepTimeout\\
  - __ecdsaWalletHeartbeatFailedCallback\\
  - notifyCloseableWallet\\
  - submitFraudChallenge\\
  - defeatFraudChallenge\\
  - defeatFraudChallengeWithHeartbeat\\
  - notifyFraudChallengeDefeatTimeout}
```

Reply:
1 - ff44cb32a3999d6b98266de5226821b495b2da43
2,3,4 - 929cc9d28be0a87dccc1a0a302e10f7ecb968e17
5 - 5809abf8b408aa7bb1a7478068ff1616c5270b89

Reply:
`notifyMovedFundsSweepProof` should be before `notifyMovedFundsSweepTimeout`. They are now reversed in the diagram code and on the generated picture.

Reply:
d2fcfd1d890c31f74abae459a1c57c02a7f426e1